rust/metrics-helper/src/metrics.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use once_cell::sync::Lazy;
+use prometheus::{
+    register_gauge, register_int_counter_vec, register_int_gauge, register_int_gauge_vec, Gauge,
+    IntCounterVec, IntGauge, IntGaugeVec,
+};
+
+/// Task failure count.
+pub static TASK_FAILURE_COUNT: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "indexer_processors_post_processing_task_failure_count",
+        "Task failure count.",
+        &["task_name"],
+    )
+    .unwrap()
+});
+
+pub static HASURA_API_LATEST_VERSION: Lazy<IntGaugeVec> = Lazy::new(|| {
+    register_int_gauge_vec!("hasura_api_latest_version", "Processor latest version", &[
```

Comment:
nit: I'd give most of these metrics a relevant prefix letting the user know that these are indeed coming from the indexer_metrics / post-processors. It will be difficult to discern on the TSDB side whether this latency is coming from hasura itself or something else like this service. see https://prometheus.io/docs/practices/naming/